### PR TITLE
Minor improvements in plot_demo_plotting example

### PR DESCRIPTION
**Pull request by KamalakerDadi**
Integrating of plotting tutorial into example ```01_plotting/plot_demo_plotting.py```

### CODE REVIEW

File: examples/01_plotting/plot_demo_plotting.py [hunk 1]
```diff
@@ -1,19 +1,22 @@
 """
-Plotting in nilearn
+Plotting tools in nilearn
 ==========================
 
-Nilearn comes with a set of plotting function for Nifti-like images,
-see :ref:`plotting` for more details.
-"""
+Nilearn comes with a set of plotting functions for an easy visualization of
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_plotting.py]: for easy visualization

File: examples/01_plotting/plot_demo_plotting.py [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
+Nifti-like images types such as statistical maps overlayed on anatomical
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_plotting.py]: Nifti-like images, such as

File: examples/01_plotting/plot_demo_plotting.py [hunk 3]
```diff
@@ -35,28 +38,50 @@
...
 plotting.plot_stat_map(localizer_tmap_filename, bg_img=localizer_anat_filename,
                        threshold=3, title="plot_stat_map",
-                       cut_coords=(36, -27, 66))
+                       cut_coords=[36, -27, 66])
 
 ###############################################################################
-# Plotting glass brain
+# Plotting statistical maps on glass brain image using function `plot_glass_brain`
+
+# Now, t-map image is overlayed on glass brain image as fixed background
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_plotting.py]: the t-map image

File: examples/01_plotting/plot_demo_plotting.py [hunk 4]
```diff
@@ -35,28 +38,50 @@
...
                   title="plot_roi")
 
 ###############################################################################
-# Plotting EPI haxby
+# Plotting EPI image using function `plot_epi`
+
+# Import image processing tool
+from nilearn import image
+
+# compute mean of the functional image from haxby dataset in t dimension.
```
Comment (bthirion) [inline examples/01_plotting/plot_demo_plotting.py]: Compute the mean functional imager

**Comment by bthirion** [comment]
This looks good, but there are small glitches to fix.

**Comment by KamalakerDadi** [comment]
Thanks for reviewing. Comments addressed.

**Comment by KamalakerDadi** [comment]
Can I have reviews on this ?

File: examples/01_plotting/plot_demo_plotting.py [hunk 5]
```diff
@@ -1,19 +1,22 @@
...
-Plotting in nilearn
+Plotting tools in nilearn
 ==========================
 
-Nilearn comes with a set of plotting function for Nifti-like images,
-see :ref:`plotting` for more details.
-"""
+Nilearn comes with a set of plotting functions for easy visualization of
+Nifti-like images such as statistical maps overlayed on anatomical
+images or overlayed on glass brain image, anatomical images, functional/EPI
```
Comment (AlexandreAbraham) [inline examples/01_plotting/plot_demo_plotting.py]: `projected on glass brain structure` ?

File: examples/01_plotting/plot_demo_plotting.py [hunk 6]
```diff
@@ -1,19 +1,22 @@
...
+See :ref:`plotting` for more details.
+"""
 
 ###############################################################################
-# Retrieve the data: haxby dataset to have EPI images and masks, and
-# localizer dataset to have contrast maps
+# First, we retrieve the data from nilearn: haxby dataset to have EPI images
+# and masks, and localizer dataset to have contrast maps
 
+# Import datasets module
```
Comment (AlexandreAbraham) [inline examples/01_plotting/plot_demo_plotting.py]: The line of code is self-explanatory, I would not add a comment.

File: examples/01_plotting/plot_demo_plotting.py [hunk 7]
```diff
@@ -35,28 +38,50 @@
...
 plotting.plot_stat_map(localizer_tmap_filename, bg_img=localizer_anat_filename,
                        threshold=3, title="plot_stat_map",
-                       cut_coords=(36, -27, 66))
+                       cut_coords=[36, -27, 66])
 
 ###############################################################################
-# Plotting glass brain
+# Plotting statistical maps on glass brain image using function `plot_glass_brain`
+
+# Now, the t-map image is overlayed on glass brain image as fixed background
```
Comment (AlexandreAbraham) [inline examples/01_plotting/plot_demo_plotting.py]: I would say projected again

**Comment by AlexandreAbraham** [comment]
Apart from some nitpicking, +1

File: examples/01_plotting/plot_demo_plotting.py (hunk 5 above)
Comment (banilo) [inline examples/01_plotting/plot_demo_plotting.py]: alternative: "mapped onto glass brain representation"

Comment (AlexandreAbraham) [inline examples/01_plotting/plot_demo_plotting.py]: Yeah, that's better!

File: examples/01_plotting/plot_demo_plotting.py [hunk 8]
```diff
@@ -1,19 +1,22 @@
...
 
-# Import plotting and image processing tools
-from nilearn import plotting, image
+See :ref:`plotting` for more details.
+"""
 
 ###############################################################################
-# Retrieve the data: haxby dataset to have EPI images and masks, and
-# localizer dataset to have contrast maps
+# First, we retrieve the data from nilearn: haxby dataset to have EPI images
```
Comment (banilo) [inline examples/01_plotting/plot_demo_plotting.py]: the data does not exactly belong to nilearn. perhaps "retrieve from nilearn-provided (general-purpose) datasets"

File: examples/01_plotting/plot_demo_plotting.py [hunk 9]
```diff
@@ -35,28 +38,50 @@
...
 
 ###############################################################################
-# Plotting EPI haxby
+# Plotting EPI image using function `plot_epi`
+
+# Import image processing tool
+from nilearn import image
+
+# Compute the mean functional image from haxby dataset in t dimension.
+# Basically reducing the functional image from 4D to 3D
```
Comment (banilo) [inline examples/01_plotting/plot_demo_plotting.py]: Alternative: "Compute the voxel-wise mean of activity maps across the time-series of maps"

**Comment by KamalakerDadi** [comment]
@AlexandreAbraham @banilo comments addressed. Thanks

**Comment by AlexandreAbraham** [comment]
Still one hanging.

File: examples/01_plotting/plot_demo_plotting.py (hunk 9 above)
Comment (KamalakerDadi) [inline examples/01_plotting/plot_demo_plotting.py]: activity maps ?

**Comment by AlexandreAbraham** [comment]
I addressed the last one in master. Thanks!

**Pull request merged**